lua/core/commands.lua

```diff
@@ -10,6 +10,25 @@ M.defaults = {
     endif
   endfunction
   ]],
+
+  -- Save files
+  -- Save file without formatting the file
+  [[command! -bang -bar SaveWithoutFormatting lua require 'utils'.save_file(false, "<bang>" == "!")]],
+  -- Save file and also format the file
+  [[command! -bang -bar SaveWithFormatting  lua require 'utils'.save_file(true, "<bang>" == "!")]],
+  -- Save file but respect auto format settings
+  [[command! -bang -bar SaveFile lua require 'utils'.save_file(nil, "<bang>" == "!")]],
+
+  -- Like above command but also quit immediately
+  [[command! -bang -bar SaveWithoutFormattingAndQuit lua require 'utils'.save_file(false, "<bang>" == "!", true)]],
+  [[command! -bang -bar SaveWithFormattingAndQuit  lua require 'utils'.save_file(true, "<bang>" == "!", true)]],
+  [[command! -bang -bar SaveFileAndQuit lua require 'utils'.save_file(nil, "<bang>" == "!", true)]],
+
+  -- Map vim commands to our new commands
+  [[cnoreabbrev w SaveFile]],
```

Comment:
can  we not change the default neovim behavior, and only use
`<leader>w` or `<leader>q` or at least make this configurable?

Reply:
A lot of people still use `:w` to save the file! If we would not do this remap, then they would not be able to format the file on save using `:w` but will have to type `:SaveFile`

But I guess that's their problem for not using `<leader>w`

But I can add and option for the user so that they can decide whether they want to the remap to be there or not! Should I?

Reply:
let chris and others comment on these, don't change anything yet :)